connector-sdk/document/src/main/java/io/camunda/operation/impl/CreateLinkOperation.java

```diff
@@ -14,19 +14,21 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.document.operation;
+package io.camunda.operation.impl;
 
 import io.camunda.document.Document;
+import io.camunda.operation.IntrinsicOperation;
+import io.camunda.operation.IntrinsicOperationProvider;
+import java.time.Duration;
+import javax.annotation.Nullable;
 
-public class Base64OperationExecutor implements DocumentOperationExecutor {
+public class CreateLinkOperation implements IntrinsicOperationProvider {
 
-  @Override
-  public boolean matches(DocumentOperation operationReference) {
-    return "base64".equalsIgnoreCase(operationReference.name());
-  }
-
-  @Override
-  public Object execute(DocumentOperation operationReference, Document document) {
-    return document.asBase64();
+  @IntrinsicOperation(name = "createLink")
+  public String execute(Document document, @Nullable Duration ttl) {
+    if (ttl != null) {
+      return document.generateLink(ttl);
+    }
+    return document.generateLink();
```

Comment:
Maybe `generateLink` could accept a `GenerateLinkRequest` that accepts a nullable TTL so we dont need two signatures.